saas/sdk/app/auth.py

```diff
@@ -271,11 +347,24 @@ def verify_password(cls, plain_password: str, hashed_password: str) -> bool:
     @classmethod
     def _authenticate_user(cls, login: str, password: str) -> Optional[User]:
         user = UserDB.get_user(login)
+        # not a user
         if not user:
             return None
 
-        if not cls.verify_password(password, user.hashed_password):
-            return None
+        # user is not disabled? verify password
+        if not user.disabled:
```

Comment:
This is problematic because if the `user` variable is assigned (which it is if this line is reached), it will return the user object without authentication in case the user is disabled. Previously, if password verification failed, `None` was returned. Please carefully check the logic because this may now be a security issue.

Reply:
Perhaps password authentication should always happen first? Or perhaps immediately return None if login attempts have exceeded the limit? Also, simplify the flow. Ideally there shouldn't be that many return statements. Make it difficult to track. If a `None` return used to indicate a wrong password, then this may have to be replaced with throwing an exception which would make it less ambiguous.